akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/LoggingEventFilter.scala

```diff
@@ -1,473 +1,188 @@
    */
-  def assertDone(max: Duration): Unit =
-    assert(
-      awaitDone(max),
-      if (todo > 0) s"$todo messages outstanding on $this"
-      else s"received ${-todo} excess messages on $this")
+  def withMessageRegex(newMessageRegex: String): LoggingEventFilter
 
   /**
-   * Apply this filter while executing the given code block. Care is taken to
-   * remove the filter when the block is finished or aborted.
+   * Matching events with an included `throwable` that is a class or subclass of the given
+   * `Throwable` `ClassTag`.
    */
-  def intercept[T](code: => T)(implicit system: ActorSystem[_]): T = {
-    interceptLogger("")(code)
-  }
+  def withCause[A <: Throwable: ClassTag]: LoggingEventFilter
 
   /**
-   * Apply this filter while executing the given code block. Care is taken to
-   * remove the filter when the block is finished or aborted.
+   * Matching events with MDC containing all entries of the given `Map`.
+   * The event MDC may have more entries than the given `Map`.
+   */
+  def withMdc(newMdc: Map[String, String]): LoggingEventFilter
+
+  /**
+   * Matching events for which the supplied function returns`true`.
    */
-  def interceptLogger[T](loggerName: String)(code: => T)(implicit system: ActorSystem[_]): T = {
-    TestAppender.setupTestAppender(loggerName)
-    TestAppender.addFilter(loggerName, this)
-    // FIXME #26537 other setting for leeway
-    import akka.actor.typed.scaladsl.adapter._
-    val leeway = TestKitExtension(system.toUntyped).TestEventFilterLeeway
-    try {
-      val result = code
-      if (!awaitDone(leeway))
-        if (todo > 0)
-          throw new AssertionError(s"timeout ($leeway) waiting for $todo messages on $this")
-        else
-          throw new AssertionError(s"received ${-todo} excess messages on $this")
-      result
-    } finally {
-      TestAppender.removeFilter(loggerName, this)
-    }
-  }
+  def withCustom(newCustom: Function[LoggingEvent, Boolean]): LoggingEventFilter
 
-  /*
-   * these default values are just there for easier subclassing
+  /**
+   * @return `true` if the event matches the conditions of the filter.
    */
-  protected val source: Option[String] = None
-  protected val message: Either[String, Regex] = Left("")
-  protected val complete: Boolean = false
+  def matches(event: LoggingEvent): Boolean
 
   /**
-   * internal implementation helper, no guaranteed API
+   * Apply this filter while executing the given code block. Care is taken to
+   * remove the filter when the block is finished or aborted.
    */
-  protected def doMatch(src: String, msg: Any) = {
-    val msgstr = if (msg != null) msg.toString else "null"
-    (source.isDefined && source.get == src || source.isEmpty) &&
-    (message match {
-      case Left(s)  => if (complete) msgstr == s else msgstr.startsWith(s)
-      case Right(p) => p.findFirstIn(msgstr).isDefined
-    })
-  }
+  def intercept[T](code: => T)(implicit system: ActorSystem[_]): T
```

Comment:
Should this Scaladoc also detail that it uses a timeout from config and that it fails if the filter is not satisfied within that timeout?

Reply:
done